src/store/filestore.rs

```diff
@@ -11,36 +12,29 @@ use std::{
     fs,
     path::{Path, PathBuf},
 };
-
-use super::{ModelID, ModelInfo, Store};
-
-const SPEC_FILE_NAME: &str = "spec.yaml";
-
-#[derive(PartialEq, Eq, PartialOrd, Ord)]
-struct NameVerTreeKey {
-    name: String,
-    version: String,
-}
-
-/// Local storage system for orca items implmenting store
+/// Support for a storage backend on a local filesystem directory.
 #[derive(Debug)]
 pub struct LocalFileStore {
+    /// A local path to a directory where store will be located.
     directory: PathBuf,
 }
 
 impl Store for LocalFileStore {
     fn save_pod(&self, pod: &Pod) -> Result<()> {
-        self.save_model(pod, &pod.hash, pod.annotation.as_ref())
+        self.save_model(
+            pod,
+            &pod.hash,
+            pod.annotation
+                .as_ref()
+                .ok_or_else(|| OrcaError::from(Kind::MissingAnnotationOnSave))?,
+        )
     }
 
     fn load_pod(&self, model_id: &ModelID) -> Result<Pod> {
-        self.load_model::<Pod>(model_id)
+        self.load_model(model_id)
     }
 
-    /// Return Btree where key is column name and value is a vec of values
-    /// Are we okay with returning a bunch of strings? For now it works but later on like adding version
-    /// this will break...
-    fn list_pod(&self) -> Result<Vec<ModelInfo>> {
+    fn list_pod(&self) -> Result<BTreeMap<String, Vec<String>>> {
```

Comment:
Thought we agree to change it to a struct?

Reply:
We did and it has been implemented using a struct `ModelInfo` internally. I think it is important to distinguish between the structure it is parsed internally and the structure that returns to the user. They could be the same but don't necessarily need to be. One is optimized for flexibility/traversal/lookups while the other is optimized for displaying in a table-friendly format.

Reply:
@guzman-raphael Change it back to model_info

Reply:
Update has been pushed.